modules/openapi-generator/src/main/resources/kotlin-client/libraries/multiplatform/infrastructure/ApiClient.kt.mustache

```diff
@@ -174,10 +158,10 @@ import {{packageName}}.auth.*
             if (requestConfig.method in listOf(RequestMethod.PUT, RequestMethod.POST, RequestMethod.PATCH))
                 this.body = body
 
-        }.response
+        }
     }
 
-    private fun RequestConfig.updateForAuth(authNames: kotlin.collections.List<String>) {
+    private fun <T: Any?> RequestConfig<T>.updateForAuth(authNames: kotlin.collections.List<String>) {
         for (authName in authNames) {
             val auth = authentications?.get(authName) ?: throw Exception("Authentication undefined: $authName")
             auth.apply(query, headers)
```

Comment:
A comment to line which mentions https://github.com/ktorio/ktor/issues/851 and the below workaround (I cannot place a comment there :disappointed:): how about removing this workaround? I see that this issue was resolved 2 years ago, so the newest version of Ktor almost for sure doesn't have this issue. On the other hand, to keep the number of changes minimal, I'm fine with doing it as a follow-up.